### DISCUSSION TITLE
Help with credential issue?

### DISCUSSION THREAD

**Discussion by glrotate** [ecederstrand/exchangelib#1050] [Q&A]
Just getting started and having an authentication issue. 

```python
from exchangelib import Credentials, Account, Configuration, DELEGATE

cred = Credentials('abc@def.gh', 'abc')

config = Configuration(server='outlook.office365.com', credentials=cred)

account = Account(primary_smtp_address='abc@def.gh', config=config, autodiscover=False, access_type=DELEGATE)

for item in account.inbox.all().order_by('-datetime_received')[:100]:
    print(item.subject, item.sender, item.datetime_received)
```

I get:

```
exchangelib\protocol.py", line 689, in raise_response_errors raise UnauthorizedError('Invalid credentials for %s' % response.url)
exchangelib.errors.UnauthorizedError: Invalid credentials for https://outlook.office365.com/EWS/Exchange.asmx
```

FWIW In a browser, https://outlook.office365.com/EWS/Exchange.asmx brings up an authentication request and the credentials don' work there either.

Is there a setting in Admin I missed?

Thanks.

**Comment by ecederstrand** [comment]
It could be that the account hasn't been granted access to use the EWS API. It's also possible that you need to set the authentication type explicitly, e.g. Basic auth, NTLM or OAuth, depending on what your organisation supports.

**Comment by socialrealm** [comment]
I've been searching for this answer for hours.  Did anybody have a solution?  Everything was working a few days ago and then it stopped and I can't seem to get it working again.

**Reply by ecederstrand** [reply]
Is this an account on Office365? Then see https://github.com/ecederstrand/exchangelib/discussions/1124#discussioncomment-3802885

**Reply by socialrealm** [reply]
Unfortunately, 
The permission needed in the link provided in your link that shows the need to add a legacy API permissions is no longer available.  This is the permission I am missing to give my token access.

![image](https://user-images.githubusercontent.com/107563270/195982918-7a98ee13-d193-4f32-8b17-a6ec6dd2b16a.png)

**Comment by faisaltechvantage** [comment]
I tried to login microsoft account that provided on code using browser. Then it logged in. After that I closed. Then again checked python code for accessing mailbox using exchangelib. It worked. I think it taking any session or something. Before that it gives error like exchangelib.errors.UnauthorizedError: Invalid credentials for https://outlook.com/autodiscover/autodiscover.svc
